python/taichi/lang/matrix.py

```diff
@@ -821,8 +821,26 @@ def field(cls,
         self.n = n
         self.m = m
         self.dt = dtype
-        for i in range(n * m):
-            self.entries.append(impl.field(dtype))
+
+        if isinstance(dtype, (list, tuple, np.ndarray)):
+            # set different dtype for each element in Matrix
+            # see #2135
+            if m == 1:
+                assert len(np.shape(dtype)) == 1 and len(
+                    dtype
+                ) == n, f'Please set correct dtype list for Vector, the shape of dtype list should be ({n}, ) not {np.shape(dtype)}'
```

Comment:
```suggestion
                ) == n, f'Please set correct dtype list for Vector. The shape of dtype list should be ({n}, ) instead of {np.shape(dtype)}'
```